src/affine.rs

```diff
@@ -114,6 +114,72 @@ impl Affine {
         let p11 = self * Point::new(rect.x1, rect.y1);
         Rect::from_points(p00, p01).union(Rect::from_points(p10, p11))
     }
+
+    /// Compute the singular value decomposition of the linear transformation (ignoring the
+    /// translation).
+    ///
+    /// All non-degenerate linear transformations can be represented as
+    ///
+    ///  1. a rotation about the origin.
+    ///  2. a scaling along the x and y axes
+    ///  3. another rotation about the origin
+    ///
+    /// composed together. Decomposing a 2x2 matrix in this way is called a "singular value
+    /// decomposition" and is written `U Σ V^T`, where U and V^T are orthogonal (rotations) and Σ
+    /// is a diagonal matrix (a scaling).
+    ///
+    /// Since currently this function is used to calculate ellipse radii and rotation from an
+    /// affine map on the unit circle, we don't calculate V^T, since a rotation of the unit (or
+    /// any) circle about its center always results in the same circle. This is the reason that an
+    /// ellipse mapped using an affine map is always an ellipse.
+    ///
+    /// Will return NaNs if the matrix (or equivalently the linear map) is singular.
+    ///
+    /// First part of the return tuple is the scaling, second part is the angle of rotation (in
+    /// radians)
+    #[inline(always)]
+    pub(crate) fn svd(self) -> (Vec2, f64) {
+        // I've put inline(always) here because we want to give the compiler the opportunity to
```

Comment:
This should just be `#[inline]`; the compiler applies heuristics for exactly the sort of thing you mention, so `always` should be reserved for cases where there is empirical evidence the compiler is missing an optimization and it's impacting performance.